--- a/drivers/net/wireless/intel/iwlwifi/pcie/trans.c
+++ b/drivers/net/wireless/intel/iwlwifi/pcie/trans.c
@@ -3793,6 +3793,9 @@ iwl_trans_pcie_alloc(struct pci_dev *pde
 {
 	struct iwl_trans_pcie *trans_pcie, **priv;
 	struct iwl_trans *trans;
+#if LINUX_VERSION_IS_LESS(6,13,0)
+	void __iomem * const *table;
+#endif
 	unsigned int bc_tbl_n_entries;
 	int ret, addr_size;
 	u32 bar0;
@@ -3926,13 +3929,23 @@ iwl_trans_pcie_alloc(struct pci_dev *pde
 		}
 	}
 
+#if LINUX_VERSION_IS_GEQ(6,13,0)
 	ret = pcim_request_all_regions(pdev, DRV_NAME);
+#else
+	ret = pcim_iomap_regions_request_all(pdev, BIT(0), DRV_NAME);
+#endif
 	if (ret) {
 		dev_err(&pdev->dev, "Requesting all PCI BARs failed.\n");
 		goto out_no_pci;
 	}
 
+#if LINUX_VERSION_IS_GEQ(6,13,0)
 	trans_pcie->hw_base = pcim_iomap(pdev, 0, 0);
+#else
+	table = pcim_iomap_table(pdev);
+	if (table)
+		trans_pcie->hw_base = table[0];
+#endif
 	if (!trans_pcie->hw_base) {
 		dev_err(&pdev->dev, "Could not ioremap PCI BAR 0.\n");
 		ret = -ENODEV;
