--- a/drivers/net/wireless/ath/ath11k/mhi.c
+++ b/drivers/net/wireless/ath/ath11k/mhi.c
@@ -348,7 +348,9 @@ int ath11k_mhi_register(struct ath11k_pc
 	ab_pci->mhi_ctrl = mhi_ctrl;
 	mhi_ctrl->cntrl_dev = ab->dev;
 	mhi_ctrl->regs = ab->mem;
+#if LINUX_VERSION_IS_GEQ(5,15,0)
 	mhi_ctrl->reg_len = ab->mem_len;
+#endif
 
 	if (ab->fw.amss_data && ab->fw.amss_len > 0) {
 		/* use MHI firmware file from firmware-N.bin */
