--- a/drivers/net/wireless/marvell/mwifiex/main.c
+++ b/drivers/net/wireless/marvell/mwifiex/main.c
@@ -1650,7 +1650,11 @@ static void mwifiex_probe_of(struct mwif
 
 	ret = devm_request_irq(dev, adapter->irq_wakeup,
 			       mwifiex_irq_wakeup_handler,
+#ifndef IRQF_NO_AUTOEN
+			       IRQF_TRIGGER_LOW,
+#else
 			       IRQF_TRIGGER_LOW | IRQF_NO_AUTOEN,
+#endif
 			       "wifi_wake", adapter);
 	if (ret) {
 		dev_err(dev, "Failed to request irq_wakeup %d (%d)\n",
@@ -1658,6 +1662,9 @@ static void mwifiex_probe_of(struct mwif
 		goto err_exit;
 	}
 
+#ifndef IRQF_NO_AUTOEN
+	disable_irq(adapter->irq_wakeup);
+#endif
 	if (device_init_wakeup(dev, true)) {
 		dev_err(dev, "fail to init wakeup for mwifiex\n");
 		goto err_exit;
